### PR TITLE
Add disabling of autoinstall globally via environment variable

### PR DESCRIPTION
**Pull request by cacheflow**
# ↪️ Pull Request
-Allow disabling of autoinstall via a environment variable named PARCEL_NO_AUTOINSTALL.

## ✔️ PR Todo

- [x ] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [ x] Included links to related issues/PRs

Feature Request: https://github.com/parcel-bundler/parcel/issues/2081

### CODE REVIEW

**Comment by mischnic** [comment]
Maybe this is a better place for disabling autoinstall:

https://github.com/parcel-bundler/parcel/blob/e86ddf45b9a46d9035dbbcbfe97a9ca1cc30ef03/src/Bundler.js#L137-L140

**Comment by cacheflow** [comment]
Agreed @mischnic.

**Comment by devongovett** [comment]
Hmm why do we need an environment variable and not just use the cli flag that already exists?

**Comment by mischnic** [comment]
I don't want to add the --no-autoinstall flag to every project of mine and having this flag set implicitly (so not specified in the package.json) doesn't change the build output.

**Comment by cacheflow** [comment]
I'd like the same in my projects @devongovett.

File: src/Bundler.js [hunk 1]
```diff
@@ -135,7 +135,8 @@ class Bundler extends EventEmitter {
       detailedReport: options.detailedReport || false,
       global: options.global,
       autoinstall:
-        typeof options.autoinstall === 'boolean'
+        typeof options.autoinstall === 'boolean' &&
+        process.env.PARCEL_NO_AUTOINSTALL !== 'true'
```
Comment (devongovett) [inline src/Bundler.js]: I think it would be better if the options were positive instead of negative, so `PARCEL_AUTOINSTALL=false` should disable it

Comment (cacheflow) [reply src/Bundler.js]: That was my original thought. Brb updating.

**Comment by devongovett** [comment]
Due to move to monorepo which rewrote the git history, this PR was automatically closed. Please open a new PR against master with your changes. You can do a diff against the `1.x` branch to see them. Sorry about that!

**Pull request closed without merging**